src/android/ChromecastUtilities.java

```diff
@@ -0,0 +1,217 @@
+package acidhax.cordova.chromecast;
+
+import android.graphics.Color;
+import android.support.v4.content.ContextCompat;
+
+import com.google.android.gms.cast.MediaInfo;
+import com.google.android.gms.cast.MediaStatus;
+import com.google.android.gms.cast.MediaTrack;
+import com.google.android.gms.cast.TextTrackStyle;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.w3c.dom.Text;
+
+class ChromecastUtilities {
+
+    static String getMediaIdleReason(MediaStatus mediaStatus) {
+        switch(mediaStatus.getIdleReason()) {
+            case MediaStatus.IDLE_REASON_CANCELED:
+                return "canceled";
+            case MediaStatus.IDLE_REASON_ERROR:
+                return "error";
+            case MediaStatus.IDLE_REASON_FINISHED:
+                return "finished";
+            case MediaStatus.IDLE_REASON_INTERRUPTED:
+                return "interrupted";
+            case MediaStatus.IDLE_REASON_NONE:
+                return "none";
+            default:
+                return null;
+        }
+    }
+
+    static String getMediaPlayerState(MediaStatus mediaStatus) {
+        switch(mediaStatus.getPlayerState()) {
+            case MediaStatus.PLAYER_STATE_BUFFERING:
+                return "BUFFERING";
```

Comment:
It looks like some of these switch statements are legacy and others are new, any chance the casing could be standardized or would that break backwards compatibility?

Reply:
What did you mean with legacy and new? Is it between Cast SDK v2 and CAF?

Reply:
I was only referring to the methods you added like `subtitle type` versus the ones that existed before like `player state`, some of them are in all caps while others are lowercase. Probably fine to leave it this way.